src/profile/hooks/useCommonEditHandling.ts

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+import to from 'await-to-js';
+
+import { EditDataValue, isNewItem } from '../helpers/editData';
+import { Action } from './useProfileDataEditor';
+import { ActionHandler } from '../components/editButtons/EditButtons';
+import { RowItemProps } from '../components/multiItemEditor/MultiItemEditor';
+
+type UseActionHandlingReturnType = {
+  currentAction: Action;
+  isEditing: boolean;
+  isNew: boolean;
+  actionHandler: ActionHandler;
+};
+
+export const useCommonEditHandling = (
+  props: RowItemProps
+): UseActionHandlingReturnType => {
+  const { data, onAction } = props;
+  const isNew = isNewItem(data);
+  const [isEditing, setEditing] = useState(isNew);
```

Comment:
isNew means it is isEditing?

Reply:
No, if the item is a new one, then edit mode should always be "active".